src/components/Button/Button.tsx

```diff
@@ -68,44 +69,45 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
       rightIcon && (
         <span
           className={cn(iconClassName, styles.icon, styles.right, {
-            [styles.noText]: !text || textIsHidden
+            [styles.noText]: noText
           })}
         >
           {rightIcon}
         </span>
       )
 
+    const getAriaLabel = () => {
+      if (ariaLabelProp) return ariaLabelProp
+      else if (hasText && typeof text === 'string') return text
+      return undefined
+    }
+
     const renderText = () =>
-      !!text &&
-      !textIsHidden && (
+      hasText && (
         <span className={cn(styles.textLabel, textClassName)}>{text}</span>
       )
 
     return (
       <button
+        aria-label={getAriaLabel()}
         className={cn(
           styles.button,
           SIZE_STYLE_MAP[size || Size.MEDIUM],
           TYPE_STYLE_MAP[type || Type.COMMON],
           {
             [styles.noIcon]: !leftIcon && !rightIcon,
-            [styles.disabled]: isDisabled,
+            [styles.disabled]: disabled,
             [styles.includeHoverAnimations]: includeHoverAnimations
           },
           className
         )}
-        name={name}
-        disabled={isDisabled}
-        onClick={onClick}
-        onMouseEnter={onMouseEnter}
-        onMouseLeave={onMouseLeave}
-        onMouseUp={onMouseUp}
-        onMouseDown={onMouseDown}
+        disabled={disabled}
+        type={buttonType}
         ref={ref}
         style={{
-          minWidth:
-            minWidth && !!text && !textIsHidden ? `${minWidth}px` : 'unset'
+          minWidth: minWidth && hasText ? `${minWidth}px` : 'unset'
         }}
+        {...other}
```

Comment:
i think this is probably fine, but potentially some unintended consequences from doing this if some caller out there is passing something they really shouldn't be

Reply:
yeah i wonder.... im mostly drawing inspo from other design libraries: https://github.com/mui-org/material-ui/blob/5daaf9be5115b1cb777fb4cf3d9125baddfe5720/packages/mui-material/src/ButtonBase/ButtonBase.js#L361

Reply:
let's just keep an eye on our staging client as we roll this out

Reply:
I've seen this type of pass through in a lot of design libraries, I'm in full support - In theory, no caller can be passing something there right now they shouldn't be if they're using typescript, right? The prop wouldn't have been available.

Reply:
^ in theory, but we may have some JS callers. Just want to make sure we remember that this could hypothetically cause regression